### ISSUE TITLE
 ‘_’ seems to be a bit unexpected.

### ISSUE THREAD

**Issue by reckless-huang** [kcl-lang/kclvm-go#265]
kcode
````
apiVersion = "apps/v1"
kind = "Deployment"
metadata = {
    name = _app
    labels.app = _app
}
spec = {
    replicas = 3
    selector.matchLabels = metadata.labels
    template.metadata.labels = metadata.labels
    template.spec.containers = [
        {
            name = metadata.name
            image = "${metadata.name}:1.14.2"
            ports = [{ containerPort = 80 }]
            $protocol = "TCP"
        }
    ]
}
#_app: str = option("app") or "default"
_app = option("app", "str", False, "", "工程名")
````
output
````
apiVersion: apps/v1
kind: Deployment
metadata:
  labels: {}
spec:
  replicas: 3
  selector:
    matchLabels: {}
  template:
    metadata:
      labels: {}
    spec:
      containers:
      - image: Undefined:1.14.2
        ports:
        - containerPort: 80
        protocol: TCP
````
why _app not render as value

**Comment by Peefy** [comment]
See the KCL document. https://www.kcl-lang.io/docs/reference/lang/tour#variables 

![image](https://github.com/kcl-lang/kcl-go/assets/22744597/3f0418d3-8e5e-4cfd-8e60-108ef0570c7e)

**Comment by Peefy** [comment]
And `_app` needs to be defined before apiVersion. This feature will be released at kcl v0.9.0 See this PR https://github.com/kcl-lang/kcl/pull/1104

**Comment by reckless-huang** [comment]
> And `_app` needs to be defined before apiVersion. This feature will be released at kcl v0.9.0 See this PR [kcl-lang/kcl#1104](https://github.com/kcl-lang/kcl/pull/1104)

get it
